src/coreclr/tools/aot/ILCompiler.Build.Tasks/ComputeManagedAssembliesToCompileToNative.cs

```diff
@@ -124,12 +124,25 @@ public override bool Execute()
 
                 // Remove any assemblies whose implementation we want to come from CoreRT's package.
                 // Currently that's System.Private.* SDK assemblies and a bunch of framework assemblies.
-                if (coreRTFrameworkAssembliesToUse.Contains(Path.GetFileName(itemSpec)))
+                var assemblyName = Path.GetFileName(itemSpec);
+                if (coreRTFrameworkAssembliesToUse.Contains(assemblyName) && assemblyName != "WindowsBase.dll")
                 {
                     assembliesToSkipPublish.Add(taskItem);
                     continue;
                 }
 
+                if (assemblyName == "WindowsBase.dll")
```

Comment:
You can move this up before `if (coreRTFrameworkAssembliesToUse.Contain...` so that the name is checked just once.